test/extensions/filters/http/transformation/transformation_filter_test.cc

```diff
@@ -43,34 +50,122 @@ class TransformationFilterTest : public testing::Test {
-
-    auto &transformation = (*route_config_.mutable_request_transformation());
-    transformation.mutable_transformation_template()->mutable_body()->set_text(
-        body);
+  void initFilterWithBodyTemplate(TransformationFilterTest::ConfigType configType, std::string body) {
+    if (configType == TransformationFilterTest::ConfigType::Listener || configType == TransformationFilterTest::ConfigType::Both) {
+      auto &transformation = (*listener_config_.mutable_request_transformation());
+      transformation.mutable_transformation_template()->mutable_body()->set_text(
+          body);
+    }  
+    if ((configType == TransformationFilterTest::ConfigType::Route || configType == TransformationFilterTest::ConfigType::Both)) {
+      auto &transformation = (*route_config_.mutable_request_transformation());
+      transformation.mutable_transformation_template()->mutable_body()->set_text(
+          body);
+    }
     initFilter(); // Re-load config.
   }
 
-  void initFilterWithBodyPassthrough() {
+  void initFilterWithBodyPassthrough(TransformationFilterTest::ConfigType configType) {
+    if (configType == TransformationFilterTest::ConfigType::Listener || configType == TransformationFilterTest::ConfigType::Both) {
+      auto &transformation = (*listener_config_.mutable_request_transformation());
+      transformation.mutable_transformation_template()->mutable_passthrough();
+    }
+    if ((configType == TransformationFilterTest::ConfigType::Route || configType == TransformationFilterTest::ConfigType::Both)) {
+      auto &transformation = (*route_config_.mutable_request_transformation());
+      transformation.mutable_transformation_template()->mutable_passthrough();
+    }
+    initFilter(); // Re-load config.
+  }
 
-    auto &transformation = (*route_config_.mutable_request_transformation());
-    transformation.mutable_transformation_template()->mutable_passthrough();
+  void initFilterWithHeadersBody(TransformationFilterTest::ConfigType configType) {
+    if (configType == TransformationFilterTest::ConfigType::Listener || configType == TransformationFilterTest::ConfigType::Both) {
+      auto &transformation = (*listener_config_.mutable_request_transformation());
+      transformation.mutable_header_body_transform();
+    }
+    if ((configType == TransformationFilterTest::ConfigType::Route || configType == TransformationFilterTest::ConfigType::Both)) {
+      auto &transformation = (*route_config_.mutable_request_transformation());
+      transformation.mutable_header_body_transform();
+    }
     initFilter(); // Re-load config.
   }
 
-  void initFilterWithHeadersBody() {
+  void transformsOnHeaders(TransformationFilterTest::ConfigType configType, unsigned int val) {
+    initFilterWithBodyTemplate(configType, "solo");
 
-    auto &transformation = (*route_config_.mutable_request_transformation());
-    transformation.mutable_header_body_transform();
+    EXPECT_CALL(filter_callbacks_, addDecodedData(_, false)).Times(1);
```

Comment:
i'm not sure if EXPECT works well in a function that is not a test... but this might have changed since last time i checked

Reply:
This is called only from within a test, I believe that’s why it works